File: java/ArqLibraryClassic/ArqLibraryLayered/src/bnsit/ms/arq/library/Borrowing.java
package bnsit.ms.arq.library;

import java.util.Date;

public class Borrowing {
    private long id;
    private final User user;
    private final Book book;
    private final Date when;

    public Borrowing(User user, Book book) {
        this.user = user;
        this.book = book;
        this.when = new Date();
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getBookId() {
        return book.getId();
    }
}

File: dotnet/ArqLibrarianClassic/ArqLibrarianClassic/Borrowing/Borrowing.cs
using System;
using ArqLibrarianClassic.Library;

namespace ArqLibrarianClassic
{
    public class Borrowing
    {
        private readonly User user;
        private readonly Book book;
        private readonly DateTime when;

        public Borrowing(User user, Book book)
        {
            this.user = user;
            this.book = book;
            this.when = DateTime.Now;
        }

        public long Id { get; set; }
        public long BookId => this.book.Id;
    }
}